### ISSUE TITLE
5.5.0 throws error Could not find browser revision 809590.

### ISSUE THREAD

**Issue by elucidsoft** [puppeteer/puppeteer#6608]
There is something definitely wrong with this version.  I have a docker container that I have been using for many versions.  It has always worked.  I updated the puppeteer version to 5.5.0, only change I made.  It doesn't work, I get an exception on start every time:

```
"Could not find browser revision 809590. Run "PUPPETEER_PRODUCT=firefox npm install" or "PUPPETEER_PRODUCT=firefox yarn install" to download a supported Firefox browser binary."

```

The docker container builds itself from scratch every single time.  I use a builder container, so its definitely not caused by residual node_modules or anything like that.  I revert the version number back to 5.4.1 and everything works again.  It's very bizarre.

I have ```ENV PUPPETEER_PRODUCT=chrome``` set in the container as well as ``` product: "chrome" ``` set in launch configuration and it STILL throws this error.  My assumption is something is wrong with how it's downloading the Chrome instance into node_modules in the new release.  I tried node 14.15.0 as well with the same results.

Pretty sure this is a bug as its 100% repeatable for me by only changing the version number of the package.

**Comment by skymanrm** [comment]
The same here, please fix

**Comment by elucidsoft** [comment]
And they ignore this completely lol

**Comment by caiusCitiriga** [comment]
Same problem here... Also in node 12, so it's not the famous Node 14 bug.

**Comment by caiusCitiriga** [comment]
Found the solution by following the readme on the Docker part. But still have to run with `--no-sandbox`

**Comment by PrayLS** [comment]
how to Solve the problem？

**Comment by caiusCitiriga** [comment]
@PrayLS this Dockerfile should make it work. You can of course personalise it as you wish. The important parts are comments (1, 2, 3, 4, 5):

```Dockerfile
FROM alpine:edge

# 1
# Installs latest Chromium (85) package.
RUN apk add --no-cache \
    chromium \
    nss \
    freetype \
    freetype-dev \
    harfbuzz \
    ca-certificates \
    ttf-freefont \
    nodejs \
    yarn

# 2
# Install node & npm to be able to install the nestjs cli and puppeteer
RUN apk add --update npm

# Create the app working directory
RUN mkdir -p /usr/src/app
WORKDIR /usr/src/app

# Copy the package from the source code to the app wd
COPY package.json ./
COPY package-lock.json ./

# Set the correct italian timezone
ENV TZ=Europe/Rome
RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone

# Install the nestjs cli
RUN npm install -g @nestjs/cli

# Install the only the PROD dependenices
RUN npm install --prod

COPY . .

# Logs folders setup
RUN mkdir -pv /usr/src/app/logs/

# Prepare built application so startup will be quicker
RUN npm run prebuild
RUN npm run build

# 3
# Tell Puppeteer to skip installing Chrome. We'll be using the installed package.
ENV PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true \
    PUPPETEER_EXECUTABLE_PATH=/usr/bin/chromium-browser

# 4
# Puppeteer v5.2.1 works with Chromium 85.
RUN npm i puppeteer@5.2.1

# 5
# It's a good idea to use dumb-init to help prevent zombie chrome processes.
ADD https://github.com/Yelp/dumb-init/releases/download/v1.2.0/dumb-init_1.2.0_amd64 /usr/local/bin/dumb-init
RUN chmod +x /usr/local/bin/dumb-init

EXPOSE 3000

CMD ["node", "dist/main"]
```

hope it helps

**Comment by bboydflo** [comment]
@caiusCitiriga while your solution might work I think people want to use the chrome version that comes with puppeteer (therefore no ENV PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true \ PUPPETEER_EXECUTABLE_PATH=/usr/bin/chromium-browser)

I ended up using this image as a base https://github.com/Zenika/alpine-chrome but now I get weird snapshots from puppeteer

**Comment by bboydflo** [comment]
I had the same issue with the latest puppeteer (5.5.0) and alpine so I ended up using this image https://github.com/Zenika/alpine-chrome, I am not sure that I actually use the chrome version that comes with puppeteer@5.5.0 but I haven't changed anything on how I launch chrome (no `executablePath` defined in my `puppeteer.launch` call). I also get weird issues (only happens within alpine container) where the snapshot image contains half of the original repeated 2 times.

**Comment by Krinkle** [comment]
> `"Could not find browser revision 809590. Run "PUPPETEER_PRODUCT=firefox npm install" or "PUPPETEER_PRODUCT=firefox yarn install" to download a supported Firefox browser binary."`
>
> I have `ENV PUPPETEER_PRODUCT=chrome` set in the container […]

The browser name is an issue with the error message only and was since fixed in v7.0.2 (https://github.com/puppeteer/puppeteer/pull/6815). 

See also:
* https://github.com/puppeteer/puppeteer/issues/6560
